test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+                                 tool.getStandardFileManager(null, null, null)) {
+                        ct = (JavacTask) tool.getTask(null,
+                                fm,
+                                null,
+                                List.of("--limit-modules", moduleName, "-d", "."),
+                                null,
+                                Collections.singletonList(new JavaSource()));
+                        ct.analyze();
+                        JavacTask finalCt = ct;
+                        ct.getElements().getAllModuleElements().stream()
+                                .forEach(me -> processModuleCheck(me, finalCt, sources));
+                        if (!sb.isEmpty()) {
+                            throw new Exception(sb.toString());
+                        }
+                    }
+
+                }
+            }
+        }
+    }
+
+
+    public void persistElement(TypeElement clazz, Element element, Types types, String version) {
+        String uniqueId = getElementName(clazz, element, types);
+        classDictionary.computeIfAbsent(uniqueId,
+                i -> new IntroducedIn(null, null));
+
+        IntroducedIn introduced = classDictionary.get(uniqueId);
+
+        if (isPreview(element, uniqueId, version)) {
+            if (introduced.introducedPreview() == null) {
+                classDictionary.put(uniqueId,
+                        new IntroducedIn(version, introduced.introducedStable()));
+            }
+        } else {
+            if (introduced.introducedStable() == null) {
+                classDictionary.put(uniqueId,
+                        new IntroducedIn(introduced.introducedPreview(), version));
+            }
+        }
+    }
+
+
+    public Version checkElement(JavadocHelper javadocHelper, String uniqueId,
+                                String currentVersion, Version enclosingVersion, Element element) {
+        String comment = null;
+        try {
+            comment = javadocHelper.getResolvedDocComment(element);
+            Version sinceVersion = comment != null ? extractSinceVersion(comment) : null;
+            if (sinceVersion == null ||
+                    (enclosingVersion != null && /*TODO: only when element overrides*/enclosingVersion.compareTo(sinceVersion) > 0)) {
+                sinceVersion = enclosingVersion;
+            }
+            IntroducedIn mappedVersion = classDictionary.get(uniqueId);
+            try {
+                String realMappedVersion = isPreview(element, uniqueId, currentVersion) ?
+                        mappedVersion.introducedPreview() :
+                        mappedVersion.introducedStable();
+                checkEquals(sinceVersion, realMappedVersion, uniqueId);
+            } catch (Exception e) {
+                System.err.println("Error for " + uniqueId + " " + e.getMessage());
+            }
+            return sinceVersion;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
I would more incline to declare the exception as thrown than catch it and re-throw it, unless there's a strong reason to not declare it.